var/spack/repos/builtin/packages/aws-parallelcluster/package.py

```diff
@@ -55,6 +55,8 @@ class AwsParallelcluster(PythonPackage):
 
     depends_on('py-pyyaml@5.2', when='@2.6:2.8 ^python@3.0:3.4', type=('build', 'run'))
     depends_on('py-pyyaml@5.3.1:', when='@2.9: ^python@:2,3.5:', type=('build', 'run'))
+    depends_on('py-pyyaml@5.1.2', when='@2.6:2.8 ^python@:2,3.5:', type=('build', 'run'))
+    depends_on('py-pyyaml@5.1.2:', when='@:2.5', type=('build', 'run'))
```

Comment:
Okay, I downloaded all versions and found:
```console
$ grep -i pyyaml aws-parallelcluster-2.*/setup.py
aws-parallelcluster-2.10.0/setup.py:    "PyYAML==5.2" if sys.version_info.major == 3 and sys.version_info.minor <= 4 else "PyYAML>=5.3.1",
aws-parallelcluster-2.10.1/setup.py:    "PyYAML==5.2" if sys.version_info.major == 3 and sys.version_info.minor <= 4 else "PyYAML>=5.3.1",
aws-parallelcluster-2.10.2/setup.py:    "PyYAML==5.2" if sys.version_info.major == 3 and sys.version_info.minor <= 4 else "PyYAML>=5.3.1",
aws-parallelcluster-2.10.3/setup.py:    "PyYAML==5.2" if sys.version_info.major == 3 and sys.version_info.minor <= 4 else "PyYAML>=5.3.1",
aws-parallelcluster-2.10.4/setup.py:    "PyYAML==5.2" if sys.version_info.major == 3 and sys.version_info.minor <= 4 else "PyYAML>=5.3.1",
aws-parallelcluster-2.11.0/setup.py:    "PyYAML>=5.3.1",
aws-parallelcluster-2.11.1/setup.py:    "PyYAML>=5.3.1",
aws-parallelcluster-2.11.2/setup.py:    "PyYAML>=5.3.1",
aws-parallelcluster-2.11.3/setup.py:    "PyYAML>=5.3.1",
aws-parallelcluster-2.11.4/setup.py:    "PyYAML>=5.3.1",
aws-parallelcluster-2.5.0/setup.py:    "PyYAML>=5.1.2",
aws-parallelcluster-2.5.1/setup.py:    "PyYAML>=5.1.2",
aws-parallelcluster-2.6.0/setup.py:    "PyYAML==5.2" if sys.version_info.major == 3 and sys.version_info.minor <= 4 else "PyYAML>=5.1.2",
aws-parallelcluster-2.6.1/setup.py:    "PyYAML==5.2" if sys.version_info.major == 3 and sys.version_info.minor <= 4 else "PyYAML>=5.1.2",
aws-parallelcluster-2.7.0/setup.py:    "PyYAML==5.2" if sys.version_info.major == 3 and sys.version_info.minor <= 4 else "PyYAML>=5.1.2",
aws-parallelcluster-2.8.0/setup.py:    "PyYAML==5.2" if sys.version_info.major == 3 and sys.version_info.minor <= 4 else "PyYAML>=5.1.2",
aws-parallelcluster-2.8.1/setup.py:    "PyYAML==5.2" if sys.version_info.major == 3 and sys.version_info.minor <= 4 else "PyYAML>=5.1.2",
aws-parallelcluster-2.9.0/setup.py:    "PyYAML==5.2" if sys.version_info.major == 3 and sys.version_info.minor <= 4 else "PyYAML>=5.3.1",
aws-parallelcluster-2.9.1/setup.py:    "PyYAML==5.2" if sys.version_info.major == 3 and sys.version_info.minor <= 4 else "PyYAML>=5.3.1",
```
So I think what we actually want is:
```suggestion
    depends_on('py-pyyaml@5.3.1:', when='@2.11:', type=('build', 'run'))
    depends_on('py-pyyaml@5.3.1:', when='@2.9:2.10 ^python@:2,3.5:', type=('build', 'run'))
    depends_on('py-pyyaml@5.2', when='@2.6:2.10 ^python@3.0:3.4', type=('build', 'run'))
    depends_on('py-pyyaml@5.1.2:', when='@2.6:2.8 ^python@:2,3.5:', type=('build', 'run'))
    depends_on('py-pyyaml@5.1.2:', when='@:2.5', type=('build', 'run'))
```

Reply:
Thanks for your extra check, patch updated!